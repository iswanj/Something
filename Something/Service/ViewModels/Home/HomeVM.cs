using Service.ViewModels.Common;
using Service.ViewModels.Enums;

namespace Service.ViewModels.Home
{
    public class HomeVM : BasePageVM
    {
        public override int PageTypeId { get; set; } = (int)PageTypeEnum.Home;
    }
}
