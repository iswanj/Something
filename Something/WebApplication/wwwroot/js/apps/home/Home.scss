@import "../Global.scss";
@import "../Variables.scss";

.Home {

}