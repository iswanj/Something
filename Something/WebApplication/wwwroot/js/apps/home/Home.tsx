import * as React from "react";
import { Button } from "antd";

import "./Home.scss";
import { IHomeProps } from "./types/IHomeProps";

export const Home : React.SFC<IHomeProps> = () => {
    return <div className="Home">
        <h1>Something.</h1>
        <p>An open source project and issue management app that doesn't suck!</p>
        <div className="Home__btnHolder">
            <Button
                size="large"
                type="primary"
                href="/account/login">
                Login
            </Button>
            <Button
                size="large"
                href="/account/register">
                Register
            </Button>
        </div>
    </div>;
}
