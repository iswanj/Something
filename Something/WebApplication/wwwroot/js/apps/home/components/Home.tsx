import * as React from "react";
import { Button } from "antd";

import "./Home.scss";

export const Home : React.SFC = () => {
    return <div className="Home">
        <Button
            type="primary"
            href="/account/login">
            Login
        </Button>
        <Button
            href="/account/register">
            Register
        </Button>
    </div>;
}
