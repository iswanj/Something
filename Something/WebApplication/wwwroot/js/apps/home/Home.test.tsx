import "jest";
import * as React from "react";
import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";

import { Home } from "./Home";

describe("<Home />", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<Home />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("has expected contents on it", () => {
        const wrapper = shallow(<Home />);
        expect(wrapper.find('.Home').length).toBe(1);
        expect(wrapper.find('.Home').find('h1').length).toBe(1);
        expect(wrapper.find('.Home').find('p').length).toBe(1);
        
        const buttonWrapper = wrapper.find('.Home').find('.Home__btnHolder');
        expect(buttonWrapper.length).toBe(1);

        const buttons = buttonWrapper.find('Button');
        expect(buttons.length).toBe(2);

        const loginButton = buttons.at(0);
        const registerButton = buttons.at(1);
        
        expect(loginButton.prop('href')).toBe('/account/login');
        expect(registerButton.prop('href')).toBe('/account/register');
    });
});
