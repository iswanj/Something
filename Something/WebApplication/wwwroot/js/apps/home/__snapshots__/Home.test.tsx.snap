// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div
  className="Home"
>
  <h1>
    Something.
  </h1>
  <p>
    An open source project and issue management app that doesn't suck!
  </p>
  <div
    className="Home__btnHolder"
  >
    <Button
      block={false}
      ghost={false}
      href="/account/login"
      loading={false}
      prefixCls="ant-btn"
      size="large"
      type="primary"
    >
      Login
    </Button>
    <Button
      block={false}
      ghost={false}
      href="/account/register"
      loading={false}
      prefixCls="ant-btn"
      size="large"
    >
      Register
    </Button>
  </div>
</div>
`;
